plugins/catalog-backend/src/catalog/DatabaseEntitiesCatalog.ts

```diff
@@ -138,27 +133,31 @@ export class DatabaseEntitiesCatalog implements EntitiesCatalog {
    * @param entities Some entities
    * @param locationId The location that they all belong to
    */
-  async batchAddOrUpdateEntities(entities: Entity[], locationId?: string) {
+  async batchAddOrUpdateEntities(
+    requests: EntityMutationRequest[],
```

Comment:
Wanted to rename the method to `batchEntityMutations` or smth like that too :>. Wouldn't mind doing the renaming in a followup PR either way 😅 

Any favorite among these?

```ts
batchAddOrUpdateEntities(requests: EntityAddOrUpdateRequest[])
batchAddOrUpdateEntities(requests: EntityUpsertRequest[])

batchMutateEntities(mutations: EntityMutation[])
batchEntityMutations(mutations: EntityMutation[])

batchUpsertEntities(upserts: EntityUpsertEntities[])
```

Reply:
Defo in a followup PR.

The last one is best I think. I like the upsert word since it has a clearly defined meaning, and it is usually used in conjunction with specifying the entire target state. The "mutate" word sounds to me like you pass in mutation "commands" so to speak, not a target state.

Reply:
Yep, right now it fits well either way and we can revisit if we end up adding deletions through this